apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,408 @@
+        public SpannerDate(DateTime dateTime) : this(dateTime.Subtract(DateTime.MinValue).Days)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpannerDate"/> structure to the specified SpannerDate.
+        /// </summary>
+        /// <param name="date">The date as <see cref="SpannerDate"/>.</param>
+        public SpannerDate(SpannerDate date) : this(date._days)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpannerDate"/> structure to the specified year, month, and day in the Gregorian calendar.
+        /// </summary>
+        /// <param name="year">The year (1 through the number of years in calendar).</param>
+        /// <param name="month">The month (1 through the number of months in calendar).</param>
+        /// <param name="day">The day (1 through the number of days in <paramref name="month"/>).</param>
+        public SpannerDate(int year, int month, int day) : this(new DateTime(year, month, day))
+        {
+        }
+
+        private SpannerDate(int days)
+        {
+            if (days < 0 || days > MaxNumberOfDays)
+            {
+                throw new ArgumentOutOfRangeException(nameof(days));
+            }
+
+            _days = days;
+        }
+
+        ///<inheritdoc/>
+        public int CompareTo(SpannerDate other) => _days.CompareTo(other._days);
+
+        /// <summary>
+        /// Implementation of <see cref="IComparable.CompareTo"/> to compare two <see cref="SpannerDate"/> values.
+        /// </summary>
+        /// <remarks>
+        /// This uses explicit interface implementation to avoid it being called accidentally. The generic implementation should usually be preferred.
+        /// </remarks>
+        /// <exception cref="ArgumentException"><paramref name="obj"/> is non-null but does not refer to an instance of <see cref="SpannerDate"/>.</exception>
+        /// <param name="obj">The object to compare this value with.</param>
+        /// <returns>The result of comparing this value with <paramref name="obj"/>. <paramref name="obj"/> is null, this method returns a value greater than 0.
+        /// </returns>
+        int IComparable.CompareTo(object obj)
+        {
+            if (obj == null)
+            {
+                return 1;
+            }
+            return obj is SpannerDate date
+                ? CompareTo(date)
+                : throw new ArgumentException($"Object must be of type {nameof(SpannerDate)}", nameof(obj));
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(SpannerDate other) => CompareTo(other) == 0;
+
+        /// <inheritdoc/>
+        public override bool Equals(object obj) => obj is SpannerDate other && Equals(other);
+
+        /// <inheritdoc/>
+        public override int GetHashCode() => _days.GetHashCode();
+
+        /// <summary>
+        /// Parses a textual representation as a <see cref="SpannerDate"/>.
```

Comment:
Does Spanner itself allow multiple cultural representations? I'm tempted to make this an "invariant-only" parse...